### PR TITLE
Redesign sign-up and login pages

### PR DESCRIPTION
**Pull request by rt4914**
## What type of PR is this? (check all applicable)

- [x] Refactor

## Description

Redesign sign-up and login pages and mainly include ToC, PP and CoC links.

## Related Tickets & Documents

- Related Issue #19988

## QA Instructions, Screenshots, Recordings

**How to test?**
Simply visit, `/enter` and `/enter?state=new-user` pages. Please check all internal links too.

### Output
Screenshot 2023-09-08 at 2 37 45 PM
Screenshot 2023-09-08 at 2 37 56 PM
Screenshot 2023-09-08 at 2 38 12 PM
Screenshot 2023-09-08 at 2 38 27 PM
Screenshot 2023-09-08 at 2 38 39 PM
Screenshot 2023-09-08 at 2 38 51 PM

Screenshot 2023-09-08 at 2 47 20 PM

### UI accessibility checklist
_If your PR includes UI changes, please utilize this checklist:_
- [x] Semantic HTML implemented?
- [x] Keyboard operability supported?
- [x] Checked with [axe DevTools](https://www.deque.com/axe/) and addressed `Critical` and `Serious` issues?
- [x] Color contrast tested?

## Added/updated tests?

- [x] Yes

### CODE REVIEW

**Comment by anujbhavsar96** [comment]
@rt4914 looks great 🚀. Just a few minor adjustments.

1. The `Padding-top` for the logo is a bit higher. It should be 24px for desktop and 16px for mobile.
image

2. Icon for Google and Forem looks slightly bigger. I'll share the exact icons with you via DM.
3. On the hover of auth icons, there is a tooltip - We can remove that.
4. For mobile - Paddings should be 16px.
image

5. We can make terms, coc, and privacy statements in two lines on mobile by removing side padding.
image

Let me know if any of this looks confusing.

**Comment by rt4914** [comment]
@anujbhavsar96 Thanks for the suggestions. I have applied all the changes.
Screenshot 2023-09-08 at 4 33 30 PM

**Review by anujbhavsar96 — APPROVED** [review]
This looks great ✨

**Comment by mirie** [comment]
@Ridhwana Would you please review this one your morning tomorrow? This will unblock this other issue: https://github.com/forem/forem/issues/19989. Thanks!

File: spec/requests/notifications_spec.rb
Comment (Ridhwana) [inline spec/requests/notifications_spec.rb:49]: It seems very vague that we're testing for OR. Same for tests below.

File: app/views/devise/registrations/_registration_form.html.erb
Comment (Ridhwana) [inline app/views/devise/registrations/_registration_form.html.erb:35]: We should use the path tags here instead of the urls. i.e. privacy_path, code_of_conduct_path, etc.

**Review by Ridhwana — COMMENTED** [review]
Left some comments, I marked some as non-blocking.

File: app/assets/stylesheets/views/signin.scss
Comment (rt4914) [reply app/assets/stylesheets/views/signin.scss:81]: @anujbhavsar96 Any suggestions what can use here instead of `13.5px` ?

Comment (anujbhavsar96) [reply app/assets/stylesheets/views/signin.scss:81]: @rt4914 As discussed over the call, it should be 12px. It was my miss while reviewing the design. `13.5px` was spacing from the label. Not the whole `div` of icon+label.

File: spec/requests/notifications_spec.rb
Comment (dukegreene) [reply spec/requests/notifications_spec.rb:49]: I agree, hard to tell at a glance what appears different between a logged-in and logged-out user

**Review by dukegreene — COMMENTED** [review]
This looks good in Uffizzi and all the links connect to the right info, form, or auth provider as expected. I agree with Ridhwana that a couple test cases could be clearer, and I'm also working on refactoring some of my own issue work to use crayons utility classes per that great suggestion; I'm able to huddle or pair on the html/css things as needed.

Provider links are not working perfectly in Uffizzi and I assume it is because of the nature of the instance affecting the client info being sent to the auth providers; will test more locally this afternoon.

File: app/assets/stylesheets/views/signin.scss
Comment (rt4914) [reply app/assets/stylesheets/views/signin.scss:78]: @Ridhwana I have tried removing custom css as much possible but I was unable to use crayons-classes correctly mostly because I was unable to find any crayons which matches closely with these designs. The biggest outliers are font-size and colors. Therefore I have used utility-generated-css in html directly (similar to tailwind implementation).

Do let me know if it can be further improved.

File: app/assets/stylesheets/views/signin.scss
Comment (rt4914) [reply app/assets/stylesheets/views/signin.scss]: Applied this change. and removed letter-spacing all together as we don't need them.

File: spec/requests/notifications_spec.rb
Comment (rt4914) [reply spec/requests/notifications_spec.rb]: Here and everywhere else we have used `OR` because earlier it was either using `Continue with` or a full text like `Have a password? Continue with your email address`. Both of these represent that we are on `Login` page. To make things less confusing I have now used `By signing in` which also represents `Login` page.

Let me know if this can be improved further.

File: app/views/devise/registrations/_registration_form.html.erb
Comment (rt4914) [reply app/views/devise/registrations/_registration_form.html.erb]: Applied this change. Thanks.

**Comment by rt4914** [comment]
@Ridhwana @dukegreene 

This PR is ready for review again. Thanks.

File: app/assets/stylesheets/base/layout.scss
Comment (Ridhwana) [inline app/assets/stylesheets/base/layout.scss:41]: (nitpick, non-blocking): Could we use the utility class `bg-base-inverted` and `max-w-100` here?

File: app/views/devise/registrations/_registration_form.html.erb
Comment (Ridhwana) [inline app/views/devise/registrations/_registration_form.html.erb:70]: (nitpick, non-blocking):  Something that we've discussed in the past but I'm not sure that we've ever documented is the use of js- prepended classes for classes that we use in JavaScript. When looking at the HTML markup it makes it more obvious to know which classes our JS depends on.

Open to having an engineering roundtable discussion about this if folks don't feel similarly.



Also I'm wondering should we have a `javascript_packs_with_chunks_tag` for the JS in this file instead of inlining it?

File: spec/requests/notifications_spec.rb
Comment (Ridhwana) [inline spec/requests/notifications_spec.rb:49]: Is the page that renders on http://localhost:3000/notifications not the "login" page instead of the "signup page"?

Screenshot 2023-09-13 at 07 26 54

It also seems like both the signin(login) and signup page have that text "By signing in", is it possible to use more distinct text to signify a signin vs signup page?

Comment (Ridhwana) [reply spec/requests/notifications_spec.rb:49]: Same comment applies for the tests that are similar below.

**Review by Ridhwana — COMMENTED** [review]
mostly non-blocking nitpicks, however, I also noticed that the signup page is not rendering as per the designs any longer: 

Screenshot 2023-09-13 at 07 23 09

File: spec/requests/notifications_spec.rb
Comment (rt4914) [reply spec/requests/notifications_spec.rb:49]: @Ridhwana If we open https://dev.to/notifications in incognito mode, we see that it shows "login" page similar to this PR's implementation. From UX perspective I think its correct because if someone is visiting the `/notifications` page then mostly likely they are a dev-signed-in-user, so showing login page makes sense. Let me know if I am missing something here.

Also, the text is slightly different on both the pages,
on login page it says -> `By signing in`
on signup page it says -> `By signing up`

This is the very small difference but atleast it is very close to the type of page it is. But I do agree that if we can improve this further it will be great, unfortunately, this is the most sensible thing I was able to find which defines the page too. 

The other option can be to use the title, i.e. `Join the` for signup pages and `Welcome back` for signin/login pages.  **Any thoughts on this?**

File: spec/requests/notifications_spec.rb
Comment (Ridhwana) [inline spec/requests/notifications_spec.rb:46]: @rt4914 I'm referring to this line where the describe block says 'it rendered the **signup** page' but in the test it's actually checking that we render the signin page. Hence the mismatch.

File: spec/requests/notifications_spec.rb
Comment (Ridhwana) [reply spec/requests/notifications_spec.rb:49]: https://github.com/forem/forem/pull/20065#discussion_r1324398182

Comment (Ridhwana) [reply spec/requests/notifications_spec.rb:49]: Regarding: 

> Also, the text is slightly different on both the pages,
on login page it says -> By signing in
on signup page it says -> By signing up

This sounds good! Thanks for clarifying.

File: spec/requests/notifications_spec.rb
Comment (rt4914) [reply spec/requests/notifications_spec.rb]: This has been fixed by updating the text to `renders the signin page`

File: app/views/devise/registrations/_registration_form.html.erb
Comment (rt4914) [reply app/views/devise/registrations/_registration_form.html.erb]: I have updated the code to use `js-registration` instead of `registrations`.

File: app/assets/stylesheets/base/layout.scss
Comment (rt4914) [reply app/assets/stylesheets/base/layout.scss]: Done.

**Comment by rt4914** [comment]
@Ridhwana Thanks for thorough review. I have replied to all the comments and here is the latest screenshot.

Screenshot 2023-09-08 at 4 33 30 PM

**Review by Ridhwana — APPROVED** [review]
@rt4914 thank you so much for making these changes! ✨ This new implementation of the login and signup pages look amazing! Kudos for the attention to detail.

**Review by dukegreene — APPROVED** [review]
Nice! UI code and specs look great

**Comment by en4839** [comment]
I noticed a few issues after this update.

1. For invites only forems the terms, privacy policy and code of content links redirect to the sign up page ( There should be a way for users to access those pages )
2. For invites only forems the "By signing up you agree..." message is showing up in the sign up page even tho there is no sign up form ( This should be removed ) 
3. The logo width is set to 48px so for forems with wide logos (not square) the logo is being squished ( You should use the site icon instead )
4. The logo should be clickable and takes us back to the homepage.
5. Maybe have an option to allow us to keep the header ? Right now the header has been removed and we really liked when the header was included on login and registration pages.
6. Suggestion: The password reset page should get the same treatment as the other pages, right now it looks off (Totally different design / layout)
7. ERROR: when I switch between the login and registration page on an invite only Forem I get the following console error: `SyntaxError: Can't create duplicate variable: 'topbarElement'`

**Comment by rt4914** [comment]
@en4839 
Thanks a lot for these detailed suggestion. Apologies from my side, I indeed missed testing on private/invite-only forems. Replied in order of above list.

1. I will check this and fix this.
2. We have already started testing this, so as soon as we replicate this, we will create a fix too.
3. Yes this will get fixed.
4. I will add this too.
5. I have informed the product/design team about this and they will make the final call here based on there research. Thanks a lot for this suggestion.
6. Informed the design team and this design should be ready soon.
7. Will fix this in upcoming PR.

Most of these fixes should happen this week. 

Again thanks a lot such detailed information. Really appreciate your efforts here.

**Pull request merged**
